featurebyte/models/entity_universe.py

```diff
@@ -130,6 +131,16 @@ def __init__(self, graph: QueryGraphModel, node: Node, source_type: SourceType):
         )
         sql_node = sql_graph.build(self.node)
         self.aggregate_input_expr = sql_node.sql
+
+        op_struct = (
+            OperationStructureExtractor(graph=flat_graph)
+            .extract(node=flat_node)
+            .operation_structure_map[flat_node.name]
+        )
+        self.aggregate_input_column_dtypes = {
+            source_col.name: source_col.dtype for source_col in op_struct.source_columns
+        }
```

Comment:
Curious whether it is possible to aggregate on the derived column?

Reply:
Since line 203 is using `get`, it should be fine even if it is not exhaustive.

Reply:
It is possible to aggregate on derived columns, but the entity column cannot be a derived column. This dict is for getting the dtype of the entity column.